src/vmm/src/devices/virtio/transport/mmio.rs

```diff
@@ -239,8 +239,8 @@ impl MmioTransport {
     }
 }
 
-impl MmioTransport {
-    pub fn bus_read(&mut self, offset: u64, data: &mut [u8]) {
+impl vm_device::BusDevice for MmioTransport {
+    fn read(&mut self, base: u64, offset: u64, data: &mut [u8]) {
```

Comment:
I see that the `base` arg is never used, so maybe it is worth removing it from a trait definition?
And same question about `Option<Arc<Barrier>>` return type in `write` function.

Reply:
These come from the external crate. So I assume they are used somewhere even if we don't use them today in Firecracker.